cargo-process.el

```diff
@@ -275,10 +275,18 @@ Always set to nil if cargo-process--enable-rust-backtrace is nil"
     (let* ((metadata-text (shell-command-to-string
                            (concat (shell-quote-argument cargo-process--custom-path-to-bin)
                                    " metadata --format-version 1 --no-deps")))
-           (metadata-json (json-read-from-string metadata-text))
+           (metadata-json (cargo-json-read-from-string metadata-text))
            (workspace-root (cdr (assoc 'workspace_root metadata-json))))
       workspace-root)))
 
+(defun cargo-json-read-from-string (text)
+  "A wrapper for `json-read-from-string' with improved error reporting."
+  (condition-case nil
+      (json-read-from-string text)
+    ;; let's assume that the invalid json is an human readable error,
```

Comment:
Maybe instead of writing this as a comment, incorporate it in the docstring?

Reply:
done!